### PR TITLE
New Notation: "Japanese"

### PR DESCRIPTION
**Pull request by Chiyozel**
Here is a notation based on the japanese numeration system.
Exponents of 10000 from https://en.wikipedia.org/wiki/Japanese_numerals#Large_numbers. 

This system also uses kanji, but were omitted from the notation for more ease of use, and also because using kanji may break the game's hitboxes as well.
Japanese-speaking people passing by may suggest rewording if needed, but only for e72+ since it is more of a sentence.

How it works:
Normal system up until e72 excluded (example: 156,162,633 = **1億5616万**). Precision is a fixed 8 digits (current myriad + previous myriad).
e72 to e10000 works just like scientific albeit more literal (4.15e136 = "**4.15×10の136乗**"). Left hand value precision based on config.
e10000 onwards: same as above, but exponents use the normal system (9.95e25416 = "**9.95×10の2万5416乗**"). Precision is based on the two above

If there are any glitches please let me know here or in the AD Discord @Yukiko#1212 so I can check the problem more thoroughly.

### CODE REVIEW

**Comment by dan-simon** [comment]
It seems that numbers between 0 and 1 (for example, 0.5) are all formatted as `0undefined`, except numbers with exponent a negative multiple of 4 which are formatted as between `1undefined` and `9undefined`.

There are two simple ways to fix this that I can see, either of which I think would work without the other being necessary. Way 1 would be not defining the `formatUnder1000` method. I'm fairly sure this would cause default formatting for numbers under 1000 (note that in some cases this might involve a decimal point, for example, 43.8; I'm not sure if Japanese has another way to handle fractions), but I haven't tested it so I might be wrong. Way 2 would be changing the `jpnNotation` method to handle numbers under 1. I think the only necessary changes to `jpnNotation` are `let exponentLast = Math.max(0, Math.floor(value.exponent / 4));` (which removes the `undefined`) and changing `value.exponent % 4` to `value.exponent - 4 * exponentLast` (which makes numbers with exponent a negative multiple of 4 also be displayed as 0).

Disclaimer: I am not the maintainer of this library or an AD dev or tester, Razen will probably have additional comments.

**Comment by Chiyozel** [comment]
Japanese people usually "format" decimal values in the same way as we do (as in, 9.5 is read pretty much "nine point five") so I decided to remove the formatting for numbers under 1000. It also makes the whole section between 0 and e4 consistent, since they display whole numbers (and decimals, for numbers below e3).

I also amended `let exponentLast = Math.max(0, Math.floor(value.exponent / 4));` in the possible event that numbers under e3 somehow get to this bit of code.

**Comment by dan-simon** [comment]
Indeed it works now, thanks for the fix.

**Comment by dan-simon** [comment]
Added in #44 as a community notation.
